custom_components/tuya_local/devices/holman_wx8_irrigation_controller.yaml

```diff
@@ -124,22 +130,37 @@ secondary_entities:
         mapping:
           - scale: 10
   - entity: binary_sensor
+    name: Battery voltage
     class: battery
+    category: diagnostic
     dps:
       - id: 120
         name: sensor
-        type: base64
+        type: bitfield
         mapping:
           - mask: "10000000"
```

Comment:
The mask is hex, so this is bit 32.  As discussed in the other conversation, just using the dps_val when it is set will auto mask the value when mapping bitfield values, so assuming you meant bit 8:
```
mapping:
  - dps_val: 128
    value: false
  - value: true
```

Reply:
OK made the chages to show dps values. Seems like a strange way to do it to me. But it does seem to work.